pilot/cmd/pilot-agent/status/server.go

```diff
@@ -584,22 +593,21 @@ func getHeaderTimeout(timeout string) (time.Duration, error) {
 // scrape will send a request to the provided url to scrape metrics from
 // This will attempt to mimic some of Prometheus functionality by passing some of the headers through
 // such as accept, timeout, and user agent
-// Returns the scraped metrics as well as the response's "Content-Type" header to determine the metrics format
-func (s *Server) scrape(url string, header http.Header) ([]byte, string, error) {
+// Returns the scraped metrics reader as well as the response's "Content-Type" header to determine the metrics format
+func (s *Server) scrape(url string, header http.Header) (io.ReadCloser, context.CancelFunc, string, error) {
+	var cancel context.CancelFunc
```

Comment:
can you call cancel in this function

Reply:
Can't. See https://github.com/istio/istio/pull/40109#discussion_r931741061

Reply:
Thanks